src/act.drive.cpp

```diff
@@ -2351,3 +2352,17 @@ int calculate_vehicle_entry_load(struct veh_data *veh) {
 ACMD(stop_rigging_first) {
   send_to_char(ch, "You'll need to stop rigging by using the %s command before you can do that.\r\n", AFF_FLAGGED(ch, AFF_RIG) ? "^WRIG^n" : "^WRETURN^n");
 }
+
+process_vehicle_decay() {
+  struct room_data *room;
+  struct veh_data *veh = NULL;
```

Comment:
This is why it crashes, your veh is null and you're not populating it before you try to check its flags. You need to iterate over the global vehicle list and test the flags for each one.